### ISSUE TITLE
Option to ignore submodules when syncing

### ISSUE THREAD

**Issue by manas-kulkarni** [desktop/desktop#15758]
### The feature request

- Currently, there are problems when syncing when a submodule has conflicts

### Proposed solution

- Option in Preferences to not care about submodules when pulling/fetching would use git pull and not touch submodules
  - Up to the user how to do submodule sync

**Comment by sergiou87** [comment]
Thanks for your suggestion @manas-kulkarni !

Could you elaborate more about that problem with the submodule conflicts?

**Comment by manas-kulkarni** [comment]
Thanks for getting back.

In short, I would like an option where GitHub desktop doesn't touch the submodules. So for example, when switching branches, it won't switch to the submodule hashes on the branch.

Our repo has a bunch of submodules that we work on extensively. Switching back and forth between branches will automatically update the submodules as well. This becomes double work. I remember that GitHub Desktop used to not touch submodules before. Would be great if we can have that back as an option

**Comment by sergiou87** [comment]
@manas-kulkarni Thank you for your reply ❤️ 

What do you mean by "This becomes double work"? We'd like to understand what the problem is exactly before deciding what the solution is, in case there is a better approach (or a bug in the current implementation).

**Comment by manas-kulkarni** [comment]
In the main repo, I switch from one branch to another, it updates all the submodules to point to the hashes in the other branch. Some times we don't want this and want to keep the submodule untouched and instead show up in the diff list.
This also affects the sync speeds as it has to keep syncing recursively instead of just sync the main repo

**Comment by sergiou87** [comment]
Thank you, that's super helpful! I totally understand how speed can get affected specially in a repo with many submodules.

> Some times we don't want this and want to keep the submodule untouched and instead show up in the diff list.

Interesting… why do you prefer to keep the submodule untouched and showing up as a change in the list? Do you _need_ that for any reason in your workflow or you just prefer having it there instead of having to wait for every submodule to update?

**Comment by manas-kulkarni** [comment]
Yes. Speed is affected since each sync will sync all submodules. Sometimes we are working on just one submodule and the rest don't need to be synced
Just a matter of preference. So I think it would be great if this option gets added to Preferences

**Comment by sergiou87** [comment]
Thank you, I will bring this feedback to the team and discuss it ❤️

**Comment by manas-kulkarni** [comment]
Any update here? I am forced to use command line since everytime I change branches through GitHub Desktop, it will sync all submodules for the other branch which is slow. It would be nice to have an option where changing a branch in GitHub Desktop has same behavior as git CLI

**Comment by sergiou87** [comment]
Hey @manas-kulkarni ! 👋 

Yes, sorry, we discussed about this 2-3 weeks ago and agreed it'd be nice to add a setting to **disable** (i.e. to opt out) automatic submodule syncing. The reasoning is that enabling this will be helpful for most people who don't want to think about submodules, and it will Just Work™, and those like you who know and want to disable it will look for this option.

However, we haven't had time to work on this lately, we've been quite busy 😞

**Comment by manas-kulkarni** [comment]
Any update?

**Comment by nnsay** [comment]
The feature also is important for monorepo. In my team, we use a monorepo repo, but we use the submodule to split some sensitive some code for security. On the other hand,  it not necessary to sync all submodule to coding for developer. So please I expect the feature to be added in github desktop.

**Comment by sergiou87** [comment]
No update, sorry, we're super busy with accessibility improvements as you can see in our PRs, and now starting with summer vacations… 😕

**Comment by mcmonkey4eva** [comment]
This should not be marked `not-planned`, this is a critical failure to handle the case of repos with submodules. I've needed heavy workarounds to avoid GitHub Desktop breaking submodules, including `ignore = all` on the `.gitmodules` prevents the module from showing up on github desktop (but doesn't stop it from glitching on branch swap).

It's gotten to the point that I now have to open a CLI and use `git checkout` to change branches as GitHub Desktop will simply error out if submodules don't align between branches, with error popups like
```
error: Entry 'MySubmodule/a/file/that/wasn't/changed/at/all.cs' would be overwritten by merge. Cannot merge.
error: Submodule 'MySubmodule' could not be updated.
```
I'm not filing this as a separate bug here as the fundamental issue is simply that _github desktop should not be touching the submodule in the first place_. The details of whatever mess it makes in the process of doing something I never asked it to do are irrelevant.

---------

Please raise the priority of this issue.
If the UI option is a barrier, I think also just default not touching submodules would be much better behavior in the majority of cases. (A UI option of course wouldn't hurt for the case where somebody does actually for some reason want that.)

Or, separately handle submodules that are on a _branch_ (`main`, `dev`, whatever) vs on a _commit_ (`abc123`). If it's on a branch, it's clearly being used for development and should not be touched. If it's on a commit, it's being autotracked by a repo consumer and thus can be automanaged.

**Comment by MintyDoggo** [comment]
Same issue here, would love to see this option available in github desktop. Half of my team does not need these submodules updated and it makes things more complex for us
